scrollytelling/src/primitive.tsx

```diff
@@ -10,6 +10,14 @@ import { useIsoLayoutEffect } from "./hooks/use-iso-layout-effect";
  * Root
  * -----------------------------------------------------------------------------------------------*/
 
+// ---- Utils
+
+type RequireAtLeastOne<T, Keys extends keyof T = keyof T> =
+    Pick<T, Exclude<keyof T, Keys>> 
+    & {
+        [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>
+    }[Keys]
+
```

Comment:
New type, maybe should move it to another file. I'm not sure of what convention we're using here so let me know.